src/towncrier/test/test_create.py

```diff
@@ -172,17 +172,39 @@ def test_create_orphan_fragment(self):
         """
         runner = CliRunner()
 
+        fragments = []
         with runner.isolated_filesystem():
             setup_simple_project()
 
             self.assertEqual([], os.listdir("foo/newsfragments"))
+            os.mkdir("foo/newsfragments/subsection")
 
-            runner.invoke(_main, ["+.feature"])
-            fragments = os.listdir("foo/newsfragments")
+            result = runner.invoke(_main, ["+.feature"])
+            self.assertEqual(0, result.exit_code)
+            result = runner.invoke(
+                _main, ["subsection/+.feature"], catch_exceptions=False
+            )
+            self.assertEqual(0, result.exit_code, result.output)
+
+            for _, dirs, files in os.walk("foo/newsfragments", topdown=False):
+                for file in files:
+                    fragments.append("/".join(dirs + [file]))
```

Comment:
Given you're using os.path everywhere, this should probably be something like `os.path.join(*dirs, file)`?

Reply:
Again, the style of the other surrounding tests hard code '/', and I was trying to minimize changes - but sure!